### PR TITLE
CLN, TST: clean up and increase coverage in tslib.pyx

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -87,11 +87,6 @@ def _test_parse_iso8601(ts: str):
 
     obj = _TSObject()
 
-    if ts == "now":
-        return Timestamp.utcnow()
-    elif ts == "today":
-        return Timestamp.now().normalize()
-
```
Comment (MarcoGorelli) [inline pandas/_libs/tslib.pyx:94]: this function is only used for testing of the ISO fastpath. there's no test that has `'now'` or `'today'`, and they wouldn't test the ISO fastpath anyway, so might as well remove them

File: pandas/_libs/tslib.pyi [hunk 2]
```diff
@@ -8,7 +8,7 @@ def format_array_from_datetime(
     values: npt.NDArray[np.int64],
     tz: tzinfo | None = ...,
     format: str | None = ...,
-    na_rep: object = ...,
+    na_rep: str | float = ...,
```
Comment (MarcoGorelli) [inline pandas/_libs/tslib.pyi:11]: `format_array_from_datetime` is only called from 

https://github.com/pandas-dev/pandas/blob/7c208c8907f5ab18f807366c0c5e26ae1dbca299/pandas/core/arrays/datetimes.py#L709-L718

, where `na_rep` can't be `None`

File: pandas/tests/indexes/datetimes/test_formats.py [hunk 3]
```diff
@@ -44,6 +44,18 @@ def test_format_native_types():
...
+    expected = np.array(
+        ["2017-01-01 00:00:00.000000", "NaT", "2017-01-03 00:00:00.000000"],
+        dtype=object,
+    )
+    tm.assert_numpy_array_equal(result, expected)
+
+    # invalid format
+    result = index._format_native_types(date_format="foo")
+    expected = np.array(["foo", "NaT", "foo"], dtype=object)
+    tm.assert_numpy_array_equal(result, expected)
```
Comment (MarcoGorelli) [inline pandas/tests/indexes/datetimes/test_formats.py:57]: just increasing coverage

File: pandas/tests/tslibs/test_parse_iso8601.py [hunk 4]
```diff
@@ -18,6 +20,7 @@
         ("2011\\01\\02", datetime(2011, 1, 2)),
         ("2013-01-01 05:30:00", datetime(2013, 1, 1, 5, 30)),
         ("2013-1-1 5:30:00", datetime(2013, 1, 1, 5, 30)),
+        ("2013-1-1 5:30:00+01:00", Timestamp(2013, 1, 1, 5, 30, tz="UTC+01:00")),
```
Comment (MarcoGorelli) [inline pandas/tests/tslibs/test_parse_iso8601.py:23]: increasing coverage

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

**Pull request merged**